src/loader.js

```diff
@@ -1,20 +1,67 @@
 import Embed from './components/embed'
+import {
+  URL,
+  IS_IOS,
+  IS_AR_QUICKLOOK_CANDIDATE,
+  IS_ANDROID,
+} from './components/utils'
+import {
+  IOS_WRAPPER,
+  ANDROID_WRAPPER,
+  DESKTOP_WRAPPER,
+} from './components/wrapper'
 
 (function() {
   const arsenal = window.arsenal || {}
   const {
-    sku = null,
+    uid = null,
+    user = null,
     target = null,
-    path = 'https://arsenal-embeds.cgtrader.com/embed/',
+    viewer = null,
+    gltf = null,
+    usdz = null,
   } = arsenal
   const domTarget = document.querySelector(target)
-  const iframe = new Embed(path + sku)
 
-  // Check if target exists
-  if (!target) {
-    console.error('Target not found')
+  // Check if required viewer props are defined
+  if (!uid && !user && !viewer && !target) {
+    return console.error('Required params missing')
+  }
+
+  function pathBuilder(item) {
+    return `${URL}${user}/${uid}/${item}`
+  }
+
+  function embedIframe() {
+    const iframe = new Embed(pathBuilder('viewer'))
+
+    // Append iframe to target placeholder
+    return domTarget.appendChild(iframe)
+  }
+
+  function embedQR() {
+    // Will be done later
+    return false
+  }
+
+  function embedAR() {
+    switch (true) {
+      case (IS_IOS && IS_AR_QUICKLOOK_CANDIDATE()):
+        return IOS_WRAPPER(pathBuilder(usdz), domTarget);
+      case IS_ANDROID:
+        return ANDROID_WRAPPER(pathBuilder(gltf), domTarget, pathBuilder('viewer'));
+      default:
+        return DESKTOP_WRAPPER(pathBuilder('viewer'), domTarget);
+    }
+  }
+
+  switch (true) {
```

Comment:
this can be `switch (viewer) {`, right?

Reply:
Nope,  it's a bool switch because this is case a bool `(IS_IOS && IS_AR_QUICKLOOK_CANDIDATE())`.